test/test_LO.py

```diff
@@ -0,0 +1,131 @@
+import torch
+from torch.testing._internal.common_utils import TestCase, run_tests
+from torch.utils._pytree import tree_map
+
+from typing import Iterator, List
+import functools
+import contextlib
+
+# TODO: move this into library proper
+@contextlib.contextmanager
+def no_dispatch() -> Iterator[None]:
+    guard = torch._C._DisableTorchDispatch()
+    try:
+        yield
+    finally:
+        del guard
+
+HANDLED_FUNCTIONS_RECTANGULAR = {}
+
+def implements(torch_aten_function, HANDLES_DICT):
+    """Register a torch function override for RectangularTensor.
+
+    This decorator takes a function in the torch API as a
+    parameter. Applying this decorator to a function adds that function
+    as the registered override for the torch function passed as a
+    parameter to the decorator.
+    """
+    @functools.wraps(torch_aten_function)
+    def decorator(func):
+        HANDLES_DICT[torch_aten_function] = func
+        return func
+    return decorator
+
+class RectangularTensor(torch.Tensor):
+    handled_functions = HANDLED_FUNCTIONS_RECTANGULAR
+    __slots__ = "full_rank", "well_cond"
+
+    @staticmethod
+    def __new__(cls, elem, *args, **kwargs):
+        # The wrapping tensor (RectangularTensor) is just a meta tensor, so it
+        # doesn't hold any memory (meta tensor is generally the preferred type
+        # of tensor you want to make a subclass from)...
+        N = elem.size(0)
+        r = torch.Tensor._make_subclass(cls, elem.to('meta'), elem.requires_grad)
+        # ...the real tensor is held as an element on the tensor.
+        r.elem = elem
```

Comment:
So, in this case, I think it might actually be better to not do the wrapper pattern, and directly pass elem into torch. A few reasons for this: first, you're not really changing the behavior of the tensor, just overriding implementations of some operations with other implementations; second, if you don't wrap like you do here, the rectangular tensor will report as a "cpu" or "cuda" tensor, rather than a "meta" tensor, which seems desirable here.